### ISSUE TITLE
px4_firmware question

### ISSUE THREAD

**Issue by lygoon** [zejdajan/sbs-sem-project-2#1]
Hello sir, sorry to bother you, I want to try to ask you some questions.

   The thing is like this, I saw the project on your github: px4_firmware, then I downloaded it, tried to compile and run: make px4_sitl gazebo_fully_atuated, but encountered a problem:
     CMake Error at platforms/posix/CMakeLists.txt:80 (install):
           install DIRECTORY given no DESTINATION!
     CMake Error at platforms/posix/CMakeLists.txt:123 (install):
           install DIRECTORY given no DESTINATION!
     CMake Error at platforms/posix/CMakeLists.txt:134 (install):
           install FILES given no DESTINATION!

   Then in the corresponding place, I changed my corresponding "CATKIN_PACKAGE_SHARE_DESTINATION" to "PROJECT_NAME", and then the compilation passed. I burned it into pixhawk 4, but after I selected the "fully_actuated" rack, I restarted and found that the number of motors in QGC is only 4 instead of 6. What is the problem?

   Here are the two problems I'm having. I don't know if there is any connection between them? There is no way to submit issues in that project, so I can only try to ask you here. If you can take the time to answer my question, then thank you very much. If possible, you can use email to communicate, my email: 1461040370@qq.com.

**Comment by zejdajan** [comment]
Hi, have you tried cloning repo with "-recursive" flag?
Also, I was using PX4 with connection to ROS, so maybe later I changed something that is incompatible with PX4 itself.

**Comment by lygoon** [comment]
Yes, I added "--recursive" when cloning. This problem has troubled me for a long time, and I don't know how to solve it. Therefore, I sincerely ask for your help.

**Comment by lygoon** [comment]
Now after my modification, 6 motors can be displayed now. But another problem arises, which is in another question of mine. I changed my corresponding "CATKIN_PACKAGE_SHARE_DESTINATION" to "PROJECT_NAME", will it have any effect? Otherwise I cannot compile.

**Comment by zejdajan** [comment]
Hi, is this problem still valid? I have seen your newer thread where you write that sim. is working.

**Comment by lygoon** [comment]
The first problem above is a compilation problem: originally you changed "PROJECT_NAME" to "CATKIN_PACKAGE_SHARE_DESTINATION", but when I compiled it, the compilation failed. Then I changed my corresponding "CATKIN_PACKAGE_SHARE_DESTINATION" to "PROJECT_NAME", and it can compile and pass. The second question about the number of motors should be solved. Looking back now, is this compilation problem because I changed your modification, which caused the real machine to be out of RC control? That's another question I just opened.

**Comment by zejdajan** [comment]
These "PROJECT NAMES" were only change to work with customized ROS, I think that for raw PX4 is the thing you have done OK.

**Comment by lygoon** [comment]
OK. Thank you for your reply.
